src/output.rs

```diff
@@ -107,28 +110,90 @@ pub fn output_activedata_etl(results: CovResultIter, output_file: Option<&str>)
+                    "covered": covered,
+                    "uncovered": uncovered,
+                    "total_covered": covered.len(),
+                    "total_uncovered": uncovered.len(),
+                    "percentage_covered": covered.len() as f32 / (covered.len() + uncovered.len()) as f32,
+                },
+                "method": {
+                    "covered": orphan_covered,
+                    "uncovered": orphan_uncovered,
+                    "total_covered": orphan_covered.len(),
+                    "total_uncovered": orphan_uncovered.len(),
+                    "percentage_covered": orphan_covered.len() as f32 / (orphan_covered.len() + orphan_uncovered.len()) as f32,
+                }
+            })
         ).unwrap();
     }
 }
 
+pub fn output_fuzzmanager(results: CovResultIter, output_file: Option<&str>) {
+    let mut writer = BufWriter::new(get_target_output_writable(output_file));
+    let mut relative: FxHashMap<PathBuf, Rc<RefCell<FMDirStats>>> = FxHashMap::default();
+    let global = Rc::new(RefCell::new(FMDirStats::new("".to_string())));
+    relative.insert(PathBuf::from(""), global.clone());
+    
+    for (abs_path, rel_path, result) in results {
+        let path = if rel_path.is_relative() {
+            rel_path
+        } else {
+            abs_path
+        };
+
+        let parent = path.parent().unwrap();
+        let mut ancestors = Vec::new();
+        for ancestor in parent.ancestors() {
+            ancestors.push(ancestor);
+            if relative.contains_key(ancestor) {
+                break;
+            }
+        }
+
+        let mut prev_stats = global.clone();
+
+        while let Some(ancestor) = ancestors.pop() {
+            prev_stats = match relative.entry(ancestor.to_path_buf()) {
+                hash_map::Entry::Occupied(s) => s.get().clone(),
+                hash_map::Entry::Vacant(p) => {
+                    let mut prev_stats = prev_stats.borrow_mut();
+                    let path_tail = if ancestor == PathBuf::from("/") {
+                        "/".to_string()
+                    } else {
+                        ancestor.file_name().unwrap().to_str().unwrap().to_string()
+                    };
+                    prev_stats.dirs.push(Rc::new(RefCell::new(FMDirStats::new(path_tail))));
+                    let last = prev_stats.dirs.last_mut().unwrap();
+                    p.insert(last.clone());
+                    last.clone()
+                },
+            };
+        }
+
+        let last_line = *result.lines.keys().last().unwrap_or(&0) as usize;
+        let mut lines: Vec<i64> = vec![-1; last_line];
+        for (line_num, line_count) in result.lines.iter() {
+            unsafe {
+                *lines.get_unchecked_mut((*line_num - 1) as usize) = *line_count as i64;
+            }
+        }
+        
+        prev_stats.borrow_mut().files.push(FMFileStats::new(path.file_name().unwrap().to_str().unwrap().to_string(), lines));
```

Comment:
Maybe it'd be better to pass result.lines to FMFileStats::new, so you don't have to iterate twice over them